optuna/storages/_rdb/storage.py

```diff
@@ -444,48 +446,68 @@ def _create_new_trial(
+
+                        trial = self._get_prepared_new_trial(study_id, template_trial, session)
+
+                        if template_trial:
+                            frozen = copy.deepcopy(template_trial)
+                            frozen.number = trial.number
+                            frozen.datetime_start = trial.datetime_start
+                            frozen._trial_id = trial.trial_id
+                        else:
+                            frozen = FrozenTrial(
+                                number=trial.number,
+                                state=trial.state,
+                                value=None,
+                                values=None,
+                                datetime_start=trial.datetime_start,
+                                datetime_complete=None,
+                                params={},
+                                distributions={},
+                                user_attrs={},
+                                system_attrs={},
+                                intermediate_values={},
+                                trial_id=trial.trial_id,
+                            )
+
+                        return frozen
+                    except Exception as e:
+                        # `sqlalchemy_exc.OperationalError` will be caught within the function.
+                        # The others will be immediately propagated to the caller.
+                        error_obj = e
                         raise
-
-                n_retries += 1
-
-            if template_trial:
-                frozen = copy.deepcopy(template_trial)
-                frozen.number = trial.number
-                frozen.datetime_start = trial.datetime_start
-                frozen._trial_id = trial.trial_id
-            else:
-                frozen = FrozenTrial(
-                    number=trial.number,
-                    state=trial.state,
-                    value=None,
-                    values=None,
-                    datetime_start=trial.datetime_start,
-                    datetime_complete=None,
-                    params={},
-                    distributions={},
-                    user_attrs={},
-                    system_attrs={},
-                    intermediate_values={},
-                    trial_id=trial.trial_id,
-                )
-
-            return frozen
+            except sqlalchemy_exc.OperationalError:
+                # Note: According to SQLAlchemy specifications,
+                # `sqlalchemy_exc.OperationalError` can be raised in situations where
+                # retries are not effective (e.g., input string is too long).
+                #
+                # It is assumed here that such bugs do not exist and retries are effective.
+                # Should such bugs be present, the content of the `OperationalError` will be
+                # propagated at the end of this function, and details will be recorded in
+                # the error log.
```

Comment:
I think what we want to say is that the unavoidable `OperationalError` would be raised after 5 retries, isn't it?

Reply:
Yes. I revised the comments.